cirq-google/cirq_google/engine/engine_client_test.py

```diff
@@ -34,6 +37,44 @@ def setup_mock_(client_constructor):
     return grpc_client
 
 
+def setup_fake_quantum_run_stream_client(client_constructor):
+    grpc_client = _FakeQuantumRunStream()
+    client_constructor.return_value = grpc_client
+    return grpc_client
+
+
+class _FakeQuantumRunStream:
+    def __init__(self):
+        self.stream_request_count = 0
+        self.cancel_requests = []
+        self.responses_and_exceptions: List[engine.QuantumRunStreamResponse | BaseException] = []
+
+    def add_responses_and_exceptions(
+        self, responses_and_exceptions: List[engine.QuantumRunStreamResponse | BaseException]
+    ):
+        self.responses_and_exceptions.extend(responses_and_exceptions)
+
+    async def quantum_run_stream(
+        self, requests: AsyncIterator[engine.QuantumRunStreamRequest] = None, **kwargs
+    ) -> AsyncIterable[engine.QuantumRunStreamResponse]:
+        async def run_async_iterator():
+            async for _ in requests:
+                self.stream_request_count += 1
+                while not self.responses_and_exceptions:
+                    await asyncio.sleep(0.1)
+                response_or_exception = self.responses_and_exceptions.pop(0)
+                if isinstance(response_or_exception, BaseException):
+                    raise response_or_exception
+                yield response_or_exception
+
+        await asyncio.sleep(0.0001)
```

Comment:
Why .0001 here and .1 above? Should there be some small $\epsilon$ that's shared for these cases?

Reply:
Turns out setting it to 0 also works!